test/tls-provider.c

```diff
@@ -2687,6 +2687,10 @@ static int xor_sig_setup_md(PROV_XORSIG_CTX *ctx,
     OPENSSL_free(ctx->aid);
     ctx->aid = NULL;
     ctx->aid_len = xor_get_aid(&(ctx->aid), ctx->sig->tls_name);
+    if (ctx->aid_len <= 0) {
+        EVP_MD_free(md);
```

Comment:
Does this need an `ERR_raise` like the error return a few lines above?

Reply:
I do not think so. There will be an error raised inside the functions called from xor_get_aid() for this error case.